src/server/trace/viewer/traceModel.ts

```diff
@@ -14,24 +14,29 @@
  * limitations under the License.
  */
 
+import fs from 'fs';
+import path from 'path';
 import * as trace from '../common/traceEvents';
 import { ContextResources, ResourceSnapshot } from '../../snapshot/snapshotTypes';
-import { SnapshotStorage } from '../../snapshot/snapshotStorage';
+import { BaseSnapshotStorage, SnapshotStorage } from '../../snapshot/snapshotStorage';
 export * as trace from '../common/traceEvents';
 
 export class TraceModel {
-  contextEntries = new Map<string, ContextEntry>();
-  pageEntries = new Map<string, { contextEntry: ContextEntry, pageEntry: PageEntry }>();
+  contextEntry: ContextEntry | undefined;
+  pageEntries = new Map<string, PageEntry>();
   contextResources = new Map<string, ContextResources>();
+  private _snapshotStorage: PersistentSnapshotStorage;
+
+  constructor(snapshotStorage: PersistentSnapshotStorage) {
+    this._snapshotStorage = snapshotStorage;
+  }
 
   appendEvents(events: trace.TraceEvent[], snapshotStorage: SnapshotStorage) {
```

Comment:
nit: might as well pass events in the constructor and make contextEntry never undefined?

Reply:
will do, following up!